src/crimson/os/seastore/seastore_types.h

```diff
@@ -700,7 +709,7 @@ struct seg_paddr_t : public paddr_t {
     internal_paddr |= static_cast<u_segment_off_t>(off);
   }
 
-  paddr_t add_offset(seastore_off_t o) const {
+  paddr_t add_offset(device_off_t o) const {
     device_off_t off = get_segment_off() + o;
```

Comment:
We may need to check o is less than max value of segment_off_t?

Reply:
Hmm, it should be good as long as the `off` for `seg_paddr_t` is reasonable.